app/models/account.rb

```diff
@@ -13,6 +13,18 @@ class Account < ApplicationRecord
 
   validates :account_number, presence: true
 
+  class << self
+    def from_identity_header(identity_header)
+      account = Account.find_or_create_by(
+        account_number: identity_header.identity['account_number']
+      )
+      account.update!(
+        is_internal: identity_header.identity['user']['is_internal']
+      )
```

Comment:
Refactoring idea: define an `is_internal` method in the IdentityHeader class, and be sure to `dig` it to handle `nil` user values (i.e. cert auth). Your test changes adding in this field signaled to me that this was an issue, otherwise I wouldn't have realized it. :laughing: